gridfs_bucket.go

```diff
@@ -115,6 +99,13 @@ func (b *gridfsBucket) Exists(ctx context.Context, key string) (bool, error) {
 
 func (b *gridfsBucket) Join(elems ...string) string { return consistentJoin(elems) }
 
+// bucket returns a new GridFS bucket configured with the context timeout, if
+// it exists. This function is called by each operation that needs to read or
+// write from the bucket to avoid read and write deadline conflicts—custom
```

Comment:
I'm not sure what this sentence about custom deadlines is saying. Are there deadlines other than the context deadline in play here? What do `custom deadlines` and `other read or write operations that also require custom deadlines` refer to in this case?

Would help understanding if we can clarify this sentence a little bit. 🙏